src/eth/Web3Service.js

```diff
@@ -62,27 +92,60 @@ export default class Web3Service extends PrivateService {
     return new this._web3.eth.Contract(abi, address);
   }
 
-  async initialize(settings) {
+  subscribeNewBlocks(cb) {
+    this._blockSub = this.subscribe('newBlockHeaders').on(
+      'data',
+      blockHeader => {
+        cb(blockHeader);
+      }
+    );
+  }
+
+  unsubscribeNewBlocks() {
+    this._blockSub.unsubscribe((err, success) => {
+      if (!success) throw new Error(err);
+    });
+  }
+
+  waitForMatchingEvent(info, event, predicate = () => true, timeout = 30000) {
+    return getMatchingEvent(this._web3, info, event, timeout, predicate);
+  }
+
+  initialize(settings) {
     this.get('log').info('Web3 is initializing...');
     this._defaultEmitter = this.get('event');
 
     this._web3 = new Web3();
     this._web3.setProvider(this.get('accounts').getProvider());
 
-    // TODO: is this still necessary? it seems confusing to have methods
-    // that look like web3.eth methods but behave differently
-    this.eth = {};
     Object.assign(
-      this.eth,
-      promisifyMethods(this._web3.eth, [
-        'getAccounts',
+      this,
+      [
         'estimateGas',
-        'getBlock',
-        'sendTransaction',
+        'getAccounts',
```

Comment:
had to change this back from [what it was changed to](https://github.com/makerdao/dai.js/pull/63/files#diff-785cfb5527ab18b80ae673e64d1947ceR135) because its behavior/meaning should not change